src/pykeen/training/lcwa.py

```diff
@@ -237,3 +225,65 @@ def _check_slicing_availability(self, supports_sub_batching: bool):
             report = "This model doesn't support sub-batching and slicing is not" " implemented for this model yet."
         logger.warning(report)
         raise MemoryError("The current model can't be trained on this hardware with these parameters.")
+
+
+# note: we use Tuple[Tensor] here, so we can re-use TensorDataset instead of having to create a custom one
+class SymmetricLCWATrainingLoop(TrainingLoop[Tuple[MappedTriples], Tuple[MappedTriples]]):
+    """
+    A "symmetric" LCWA scoring heads *and* tails at once.
```

Comment:
Can we please get more high level documentation here? You already made a nice write up in the issue, just move it over here. Probably also link the issue itself